src/main/scala/mesosphere/marathon/api/v2/validation/PodsValidation.scala

```diff
@@ -239,6 +253,45 @@ trait PodsValidation extends GeneralPurposeCombinators {
         new And(plugins: _*).apply(pod)
       }
     }
+
+  def residentUpdateIsValid(from: PodDefinition): Validator[PodDefinition] = {
+    val changeNoVolumes =
+      isTrue[PodDefinition]("persistent volumes must not change") { to =>
+        val fromVolumes = from.persistentVolumes
+        val toVolumes = to.persistentVolumes
+        def sameSize = fromVolumes.size == toVolumes.size
+        def noVolumeChange = fromVolumes.forall { fromVolume =>
+          toVolumes.find(_.name == fromVolume.name).contains(fromVolume)
+        }
+        sameSize && noVolumeChange
+      }
+
+    val changeNoResource =
+      isTrue[PodDefinition]("pods with persistent volumes must not have any resource changes") { to =>
```

Comment:
will this tell the user which resource it is? I mean if I got just this message I would not immediately know what I did wrong...

Maybe we can split this validator into multiple validators?